envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/RemoteServicesTest.kt

```diff
@@ -41,6 +41,21 @@ class RemoteServicesTest {
         assertThat(result.map { it.cluster }).contains("dc1")
     }
 
+    @Test
+    fun `should ignore services without instances`() {
```

Comment:
there is a lot of implicit knowledge here hidden in the config of the `asyncClient` and the `servicesState4` – reading this test requires looking up the implementation of the function `asyncClient` and then jumping to the definition of `servicesState4`. I'd suggest using meaningful names instead of "4" for the required state.
Maybe not part of this PR but in the future I'd prefer a setup per test instead of an entire aggregate for the entire suite – I created an issue: https://github.com/allegro/envoy-control/issues/168